internal/storage/azure_blobstore.go

```diff
@@ -34,6 +36,42 @@ type AzureBlobstore struct {
 	serviceURL *azblob.ServiceURL
 }
 
+func newAccessTokenCredential(accountName string, accountKey string) (azblob.Credential, error) {
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		return nil, fmt.Errorf("storage.newAccessTokenCredential: %w", err)
+	}
+	return credential, nil
+}
+
+func newMSITokenCredential(blobstoreURL string) (azblob.Credential, error) {
+	msiEndpoint, err := adal.GetMSIVMEndpoint()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get MSI endpoint: %w", err)
+	}
+
+	spt, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, blobstoreURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get service principal token from msi %v: %v", msiEndpoint, err)
+	}
+
+	var tokenRefresher azblob.TokenRefresher
+	tokenRefresher = func(credential azblob.TokenCredential) time.Duration {
+		err := spt.Refresh()
+		if err != nil {
+			panic(err)
```

Comment:
I'm not a huge fan of panicing here. Can we return an error or maybe just log instead?

Reply:
I'm pretty new to Go so I wasn't sure if this was a good thing to do or not. @nmiodice pointed me to the documentation for this package [here](https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob). 

>  Your tokenRefresher function must return a time.Duration indicating how long the TokenCredential object should wait before calling your tokenRefresher function again. **If your tokenRefresher callback fails to refresh the token, you can return a duration of 0 to stop your TokenCredential object from ever invoking tokenRefresher again. Also, oen way to deal with failing to refresh a token is to cancel a context.Context object used by requests that have the TokenCredential object in their pipeline.**

I changed this to log the error and return a duration of 0 which will prevent the token from refreshing.

Reply:
Welcome to Go 😄. It's a fun language I promise